api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java

```diff
@@ -3,65 +3,192 @@
+    assertThat(dbInstitution).isNull();
+    assertThat(institutionDao.findAll()).containsExactly(institutionWithEmailAddressesAndDomains);
   }
 
   @Test
   public void test_findAll() {
-    assertThat(institutionDao.findAll()).containsExactlyElementsIn(ImmutableList.of(testInst));
-
-    DbInstitution otherInst = new DbInstitution("VUMC", "Vanderbilt");
-    otherInst = institutionDao.save(otherInst);
     assertThat(institutionDao.findAll())
-        .containsExactlyElementsIn(ImmutableList.of(testInst, otherInst));
+        .containsExactly(
+            institutionNoEmailAddressesOrDomains, institutionWithEmailAddressesAndDomains);
+  }
 
-    institutionDao.delete(institutionDao.findOneByShortName("Broad").get());
-    assertThat(institutionDao.findAll()).containsExactlyElementsIn(ImmutableList.of(otherInst));
+  @Test
+  public void test_findOne() {
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionNoEmailAddressesOrDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionNoEmailAddressesOrDomains);
+    assertThat(dbInstitution.getEmailDomains()).isEmpty();
+    assertThat(dbInstitution.getEmailAddresses()).isEmpty();
+
+    dbInstitution =
+        institutionDao.findOne(institutionWithEmailAddressesAndDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionWithEmailAddressesAndDomains);
+    assertThat(dbInstitution.getEmailDomains()).isEqualTo(emailDomains);
+    assertThat(dbInstitution.getEmailAddresses()).isEqualTo(emailAddresses);
   }
 
   @Test
   public void test_findOneByShortName() {
-    assertThat(institutionDao.findOneByShortName("Broad")).hasValue(testInst);
+    assertThat(institutionDao.findOneByShortName("Broad"))
+        .hasValue(institutionNoEmailAddressesOrDomains);
+    assertThat(institutionDao.findOneByShortName("Broad1"))
+        .hasValue(institutionWithEmailAddressesAndDomains);
     assertThat(institutionDao.findOneByShortName("Verily")).isEmpty();
+  }
+
+  @Test
+  public void test_updateAllNewEmailAddresses() {
+    Set<DbInstitutionEmailAddress> newEmailAddresses =
+        Sets.newHashSet(
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress1"),
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress2"));
+    institutionWithEmailAddressesAndDomains.setEmailAddresses(newEmailAddresses);
+    institutionDao.save(institutionWithEmailAddressesAndDomains);
+
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionWithEmailAddressesAndDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionWithEmailAddressesAndDomains);
+    assertThat(dbInstitution.getEmailAddresses()).containsExactlyElementsIn(newEmailAddresses);
+  }
+
+  @Test
+  public void test_updateWithExistingEmailAddresses() {
+    Set<DbInstitutionEmailAddress> newEmailAddresses =
+        Sets.newHashSet(
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress"),
```

Comment:
nit: make args look like email addresses

Also, we need some testing of the sharing (if allowed) of addresses across institutions.